### ISSUE TITLE
Prepare Release 3.8.1

### ISSUE THREAD

**Issue by ann0see** [jamulussoftware/jamulus#1972]
**Target timeline**
Scheduled feature freeze / Start of translation process: 2021-09-15
Targeted translation completion date: 2021-09-30 
Approximate release date: 2021-10-22 (actual 2021-10-23)
Current state: Released

**Checklist**
- [ ] Assign this issue to the release shepherd who is in charge of managing this checklist (I won’t since I‘ll focus on the website)
- [x] Pin this issue
- [x] Create a milestone and add all relevant issues
- [x] Ensure that all issues/PR targeted for this release are done by checking the Project board with an appropriate filter (e.g. [`milestone:"Release 3.8.0"`](https://github.com/orgs/jamulussoftware/projects/2?card_filter_query=milestone%3A"release+3.8.0")). Reminder main developers to review entries in *Waiting on team* state. De-tag unfinished Issues/PRs.
- [x] Declare a freeze for code website by updating this Issue and adding a comment. PRs can still be worked on and may get reviewed, but must not be merged unless agreed explicitly.
- [x] Check ./Jamulus -h output against the en-Command-Line-Options.md wiki page and update if necessary.
- [x] Write the Changelog based on the list of PRs in "done" state ([example for 3.8.1](https://github.com/orgs/jamulussoftware/projects/2?card_filter_query=milestone%3A%22release+3.8.1%22+is%3Amerged))
- [x] Start Website translations
  - [x] Check for broken links with a link checker on the `next-release` branch and fix them
  - [x] Check again if all issues tagged to this release are merged to `next-release`
  - [x] Check if all merged Pull Requests on the jamulussoftware/jamulus repository which needed documentation have been documented and everything is in sync
  - [x] Check for conflicts between the `release` and `next-release` branch (e.g. by opening a Pull Request from `release` to `next-release`)
  - [x] Open a Pull Request from `next-release` to release, set it as "Draft" and briefly scan the changes for typos or errors in the English version
  - [x] Declare a full freeze of the `next-release` and `release` branch. No changes should be made from now on to ensure translators don't have to work twice.
  - [x] Check if the list of translators in `tools/create-translation-issues.sh` and issue texts are up-to-date
  - [x] Create a translation issue for each language with `tools/create-translation-issues.sh` using `web` argument (see notes in script). 
  - [x] If anyone finds critical issues now, all translators must be made aware of them and all languages should be updated.
- [x] Start App translations
  - [x] Generate `.ts` files in master via `lupdate`
  - [x] Check if the list of translators in `tools/create-translation-issues.sh` is up-to-date
  - [x] Create a translation issue for each language with `tools/create-translation-issues.sh` using `app` argument.
- [x] [Tag a beta release](https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md#steps-for-a-specific-release) (inform emlynmac for signing on macOS)
- [x] Announce the beta release on Github Discussions. Pin the thread.
- [x] Get feedback on the stability and resource usage (memleaks?, crashes?, installation issues?) of the beta release
- [x] Finish Website translations
  - [x] Wait for all PRs to be merged (missing translations will revert to English automatically)
  - [x] Check for broken links with a link checker locally (run `_po4a-tools/po4a-create-all-targets.sh` locally first)
- [x] Finish App translations
  - [x] Review PRs <details><summary>Checklist for the PRs</summary> 
     ```
     - [ ] Translator listed in the `src/util.cpp` **optionally add link to PR or code**
     - [ ] Punctuation and spacing consistent
     - [ ] Signal words consistent ("ASIO", "Buffer")
     - [ ] App translations: No untranslated strings (`grep unfinished -5 src/res/translation/translation_$TRANSLATION*.ts`)
     - [ ] App translations: Only a single `.ts` file checked in (`.qm` in addition is also OK)
     - [ ] Installer translations: Passes `tools/check-wininstaller-translations.sh`
     ```
     </details>
  - [x] Wait for all PRs to be merged (missing translations will revert to English automatically). 
  - [x] Check for conflicting accelerator keys (see `tools/checkkeys.pl`)
  - [x] Generate `.qm` files via `lrelease Jamulus.pro`  
- [x] [Tag a release candidate](https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md#steps-for-a-specific-release) (inform emlynmac for signing on macOS)
- [x] Announce the release candidate on Github Discussions. Pin the thread. Unpin and lock the beta thread.
- [x] Draft an announcement, include all contributors via `tools/get_release_contributors.py`
- [x] [Update the version number in `Jamulus.pro` and add the release date to the Changelog header and commit](https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md#steps-for-a-specific-release)
- [x] Tag this commit as `r3_y_z`
- [x] Wait for the build to complete
- [x] Contact emlynmac for signing on macOS
- [x] Do a smoke test for Windows/Mac/Linux -- Do the binaries start/connect properly? Can earlier Jamulus versions properly connect to a server based on the new release?
- [x] [Force tag that tag as `latest` and push.](https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md#if-this-is-a-proper-release-move-the-latest-tag)
- [x] Upload the artifacts to SourceForge and link latest (see #1814 for docs)
- [x] [Prepare `Jamulus.pro` (`dev` suffix) and ChangeLog (add a header) for the next release](https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md#make-the-master-branch-ready-for-post-release-development)
- [x] Update download links on the website by editing `config.yml`
- [x] Publish Website release by merging `next-release` into `release`
- [x] Announce the new release with a summary of changes (+ link to the changelog for details) and a link to the download page
  - [x] On Github Discussions in the Announcements section. Lock the announcement thread. Pin the thread. Unpin and lock release candidate thread.
  - [x] On Facebook in the group "Jamulus (official group)". Turn off replies.
- [x] Trigger the update notification by updating both Update Check Servers with the new version (@pljones for update02, email corrados for update01)
- [x] Delete the `next-release` branch from the Website repo
- [x] Check that all Issues and PRs tagged for this release are in Done/Closed state.
- [x] Close the release milestone in both jamulus and jamuluswebsite repos
- [x] Create a milestone for the next minor release in jamulus and jamuluswebsite repos
- [x] Update the release checklist with any improvements which were found
- [x] Create a release retrospective Discussion
- [x] Unpin and close this issue
- [x] Update this template in https://jamulus.io/contribute/Release-Process if needed.

**Comment by JaminShanti** [comment]
What features are you considering?  I would like to review.

**Comment by pljones** [comment]
@JaminShanti if the issue has already been raised and a pull request is being considered, then it's possible it could be in the release.

Otherwise, it's missed 3.8.1.

**Comment by ann0see** [comment]
Agree. I think we should stop tagging new PRs soon and then work on the remaining tagged ones. I will not be available from the end of September due to personal reasons, so at least the we should start freezing as soon as possible.

**Comment by ann0see** [comment]
I've moved https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md into https://github.com/jamulussoftware/jamuluswebsite/pull/592
I think it makes sense to put all the release related stuff on the website and make it "copy and pastable".

**Comment by gilgongo** [comment]
I think we need to finish - or de-tag - any PRs that are [not yet finished for this release](https://github.com/orgs/jamulussoftware/projects/2?card_filter_query=milestone%3A%22release+3.8.1%22+is%3Apr):

#1794 
#1829 
#1889 
#1954 
#1927 
#1741 

BTW just looking at the release milestone in GH is not helpful in the run-up to a release becuase it doesn't show what the actual status of something is. Anything can be tagged for a release - it doesn't mean it's going in :-)

**Comment by ann0see** [comment]
@JaminShanti you can have a look at @gilgongo ‘s list from above and help out by testing the beta releases.

**Comment by JaminShanti** [comment]
I'll try out the beta today.  I don't code in C. Sry.  I do code in python , bash , and powershell if you need any help there.

**Comment by ann0see** [comment]
It’s not yet out; however I can tag a nightly which would then appear here: https://github.com/jamulussoftware/jamulus/releases

for PowerShell, the Windows build scripts might be a good read. You might want to have a look at JACK Support for Windows: https://github.com/jamulussoftware/jamulus/pull/1829 and review it.
Python is probably in the /autobuild/ folder and bash too. Actually, the PowerShell scripts might be a bit complicated. Maybe this bigger proposal of changing e.g. the build process: https://github.com/jamulussoftware/jamulus/discussions/1936 might be interesting?

OFC, you can also look into other PRs/issues ;-)

**Comment by ann0see** [comment]
Pushed a nightly: https://github.com/jamulussoftware/jamulus/releases/tag/r3.8.0devNightly2

People who want to try it need to know that it's still an early pre-release without translation etc. Please test it for bugs, regressions...

**Comment by JaminShanti** [comment]
I took a look at the windows build.  I don't know much about QT.  Seems like a big download.  For now, I'll just test out the devNightly2.

**Comment by ann0see** [comment]
Yeah, it’s big since we have 32Bit and 64Bit

**Comment by JaminShanti** [comment]
Used it for a few hours tonight, Seemed Good.  Noticed the Connect Screen seemed faster. Populating with Available Public Servers.

**Comment by pljones** [comment]
https://github.com/jamulussoftware/jamulus/commit/2ad6ef836481305bfc39b7c65772654881560b70
That looks odd.  We shouldn't be cutting builds from anything but master and this isn't master.

**Comment by pljones** [comment]
#1794 and #1741 I think we'll have to say aren't going to make it -- there's not been movement on them for too long.

#1829 looks nearly there.  #1927 looks ready?  #1954 is blocked by #1927.

#1889 is done.

So looks like we're nearly at the lock point.

**Comment by gilgongo** [comment]
#1927 looks ready for merging judging by the last comment, but it's hard to say as quite a lot of things in the conversion haven't been marked as resolved even though they look to me like they have. I can merge it (and consequently #1954) though if needed.

**Comment by gilgongo** [comment]
Let's de-tag #1794 and #1741 on Tuesday (that is, about 28 hours) unless they get some love.

**Comment by JaminShanti** [comment]
Running 2ad6ef8 tonight.  The Directory option could really use a rework.  I find myself toggling the directory to see which servers are active, maybe list everything in one screen and add a category field.  Also would be nice to have a friends list to find which severs people are on.  I know this would be some privacy concerns.  But unique identifiers would allow you to find people that you jammed with the night before.

**Comment by gilgongo** [comment]
@JaminShanti 

> maybe list everything in one screen and add a category field

Unfortunately, we can only list up to 150 servers per directory otherwise UDP packets will fragment too much (and directories would get hit by too much load). Hence the need to split directories by "genre" but there have been discussions on how to overcome this. Friends lists have been requested in the past in various forms too - see the [discussion forums](https://github.com/jamulussoftware/jamulus/discussions/1774) for feature requests in general.

You can find [compilation instructions here](https://github.com/jamulussoftware/jamulus/blob/master/COMPILING.md).

**Comment by gene96817** [comment]
Both concepts can be used together. Sounds to me like we can distribute (directory server to  client) the list of servers in 150 server chunks. The meaning of the chunks an be arbritary. On the client used it is possible to assemble the chunks into "one screen" and have a category field to select subsets. Or, just use the existing genre chunks from the directory server. The client can assemble/merge the chunks in to a single list and have category fields to select directory subsets. This implies the category fields could be the current genres or something else.

**Comment by JaminShanti** [comment]
I have notice much effort listing servers no one is connected too.  May unlist unused servers from default filter. For instance the first thing I do when I login is add # to the search filter.

**Comment by ann0see** [comment]
It depends on your usecase of Jamulus. Many ensambles search for empty servers while Jammers search for full ones. But please open a separate discussion on that.

**Comment by gilgongo** [comment]
We have one last ticket that I've found that wasn't being tracked, but seems to me to be worth getting in for this release: https://github.com/jamulussoftware/jamulus/pull/1992

**Comment by ignotus666** [comment]
There's no hurry yet for translations though is there? (re #1992). I mean, strictly speaking, it would have been better if he'd waited for the freeze and translation process to begin because there are strings that aren't in the .ts files yet.

**Comment by gilgongo** [comment]
OK next up: some issues that are tagged for this relase but are still hanging:

https://github.com/jamulussoftware/jamulus/issues/1780 - can somebody action this? 

https://github.com/jamulussoftware/jamulus/issues/1791 - looks like it needs shunting to 3.9?

https://github.com/jamulussoftware/jamulus/issues/1797 -  looks like it needs shunting to 3.9?

**Comment by softins** [comment]
> OK next up: some issues that are tagged for this relase but are still hanging:
> 
> #1780 - can somebody action this?

OK, sure

> #1791 - looks like it needs shunting to 3.9?

Yes, we haven't really begun to understand the cause yet. It's not confined to Big Sur, and I believe it's specific to the Mac port of Qt. The "hang" is not permanent, and usually clears after several seconds. It also doesn't appear to affect the audio.

> #1797 - looks like it needs shunting to 3.9?

Yes, this is not critical, just flagging up a potential future issue if dmgbuild ceases to be supported/available.

**Comment by gilgongo** [comment]
@softins OK thanks - I've remove the milestones on 1791 and 1797 for now.

**Comment by gilgongo** [comment]
OK looks like all is now merged and dealt with for this milestone. 

Code freeze for Release 3.8.1 now on.

**Comment by ann0see** [comment]
Ok. If that’s the case, we need to update the documentation and in parallel open the translation issues with hoffie‘s script? Was this PR merged already?

**Comment by ann0see** [comment]
No… https://github.com/jamulussoftware/jamulus/pull/1594 should be merged, I think

**Comment by gilgongo** [comment]
OK that's done. Anything else?

**Comment by jujudusud** [comment]
Hi everyone, is there any translations to do on the installer or the application ? (Independantly of thé documentations).

**Comment by henkdegroot** [comment]
> any translations to do

I believe there is. The. ts files in master have been updated, so you can check/start already. A few strings are different formatted and some text is new.
Guess the issues for translation will be created soon.

**Comment by gilgongo** [comment]
Last call for any updates to the English on the website. I'll be drafting the PR from next-release (and opening up tickets) later today otherwise.

**Comment by gilgongo** [comment]
**OK website full freeze of the `next-release` and `release` branches is on now. No changes should be made from now on to ensure translators don't have to work twice.**

**Comment by gilgongo** [comment]
@jamulussoftware/translators  you are all now invited to translate :slight_smile: 

Please note that if you are translating the website, there have been some quite large structural changes to that, but I think the actual text of much that was there in the last release is the same.

**Comment by gilgongo** [comment]
@jamulussoftware/translators With luck, this is early enough not to cause problems, but @ignotus discovered  a formatting error in `Installation-for-Linux.md` which I'd like to push to next-release on the website if that's OK. It's basically a new line after a header. Not having it will cause formatting problems with p4oa.

**Comment by jujudusud** [comment]
> @jamulussoftware/translators With luck, this is early enough not to cause problems, but @Ignotus discovered a formatting error in `Installation-for-Linux.md` which I'd like to push to next-release on the website if that's OK. It's basically a new line after a header. Not having it will cause formatting problems with p4oa.

Please do ! :-)
